DiffPlex/DiffBuilder/SideBySideDiffBuilder.cs

```diff
@@ -84,36 +91,49 @@ public static SideBySideDiffModel Diff(IDiffer differ, string oldText, string ne
 
             if (differ == null) return Diff(oldText, newText, ignoreWhiteSpace, ignoreCase);
             var model = new SideBySideDiffModel();
+            bool hasDifferences;
+            
             var diffResult = differ.CreateDiffs(oldText, newText, ignoreWhiteSpace, ignoreCase, lineChunker ?? LineChunker.Instance);
-            BuildDiffPieces(diffResult, model.OldText.Lines, model.NewText.Lines, (ot, nt, op, np) =>
+            BuildDiffPieces(diffResult, model.OldText.Lines, model.NewText.Lines, out hasDifferences, (ot, nt, op, np) =>
             {
                 var r = differ.CreateDiffs(oldText, newText, false, false, wordChunker ?? WordChunker.Instance);
-                BuildDiffPieces(r, op, np, null);
+                BuildDiffPieces(r, op, np, out hasDifferences, null);
```

Comment:
I don't think this would be needed to get hasDifference from the inner diffing but since the contract has it you are a bit forced. I think the same as comments above, can we just do the has difference calculation in this method itself?

Reply:
So basicly you want to move logic into `Diff` methods where `Models` are initialized?

Reply:
Yea. Do you see any issues with that?

Reply:
Its looks even easier. I thought about my `out` parameter as a little over engeenering thing but i didnt want ask too much. After your CR all is clear